src/searcher/glue.rs

```diff
@@ -0,0 +1,262 @@
+use tree_sitter::{Node, QueryCursor};
+
+use crate::{
+    lines::{self, LineStep},
+    plugin::get_loaded_filter,
+    query_context::QueryContext,
+    searcher::{core::Core, Config, Range, Searcher},
+    sink::Sink,
+    treesitter::{get_parser, node_to_match},
+};
+
+#[derive(Debug, Default)]
+struct AccumulatedExactMatches {
```

Comment:
This type and the places where we add "exact matches" to it are a little janky-feeling but the basic idea is it looked like we need to keep track of the beginning of the first line of the "clump" of match lines that are going to be the byte range that gets passed to the sink API so that we can specify the exact matches that we're now passing there as relative to that byte range